### PR TITLE
Update to crystal 0.25.0

### LINKED ISSUE #226: Get ready for Crystal 0.25.0
With 0.25.0 coming soon, we need to update some things for the new version.

- [x]  Update Dockerfile since `crystal deps` is being removed. https://github.com/crystal-lang/crystal/pull/5544
- [ ]  Remove references of `JSON::Type`. https://github.com/crystal-lang/crystal/pull/5183
- [ ] `File.stat` is renamed to `File.info` https://github.com/crystal-lang/crystal/pull/5584
- [ ]  Read the changelog for anything I missed. https://github.com/bcardiff/crystal/blob/changelog/0.25.0/CHANGELOG.md

### PR DESCRIPTION
**Pull request by faustinoaq**
Minimally addresses issues required to update Granite to Crystal 0.25.0

- [x]  Update Dockerfile since `crystal deps` is being removed. https://github.com/crystal-lang/crystal/pull/5544
- [x]  Remove references of `JSON::Type`. https://github.com/crystal-lang/crystal/pull/5183
- [x] `File.stat` is renamed to `File.info` https://github.com/crystal-lang/crystal/pull/5584
- [x] Update all instances of `Time::Kind` to use the new `Time::Location` instead

Read the changelog for anything I missed. https://github.com/bcardiff/crystal/blob/changelog/0.25.0/CHANGELOG.md

Fixes #226

### CODE REVIEW

**Comment by Blacksmoke16** [comment]
@faustinoaq We probably should also address other changes specified in https://github.com/amberframework/granite/issues/226.

File: src/granite/querying.cr [hunk 1]
```diff
@@ -23,7 +23,7 @@ module Granite::Querying
             if @@adapter.class.name == "Granite::Adapter::Sqlite"
               # sqlite3 does not have timestamp type - timestamps are stored as str
               # will break for null timestamps
-              self.\{{name.id}} = Time.parse(result.read(String), "%F %X" )
+              self.\{{name.id}} = Time.parse(result.read(String), Granite::DATETIME_FORMAT)
```
Comment (drujensen) [inline src/granite/querying.cr]: nice change to `Granite::DATETIME_FORMAT` 💯

**Comment by damianham** [comment]
why not rescue the exception for existing time strings and try to parse with "%F %X" ?

**Comment by faustinoaq** [comment]
> why not rescue the exception for existing time strings and try to parse with "%F %X" ?

Interesting idea, Would this solve the SQLite issue?

^^ @robacarp @drujensen

**Comment by drujensen** [comment]
I think this might work with one caveat:

We either need to store the time as UTC as a convention or we need to include the timezone in the stored string. You should always store the timezone implicitly or explicitly to get the right results.

I prefer we store the timezone in the string explicitly.  The advantage is you can determine the timezone that was originally used by the user/server when it was originally stored.  If you convert it to UTC before storing, you lose that information.

**Comment by robacarp** [comment]
I agree, it makes sense for Granite to store the timezone as provided in the Time object given. In the event that it makes more sense for an application timezone to always be UTC, the appdev can make that happen easy enough.

**Pull request merged**